libvast/test/address.cpp

```diff
@@ -146,3 +148,230 @@ TEST(parseable) {
+    {"::ffff", "78ff:f001:9fc0:20df:8380:b1f1:704:f838"},
+    {"2001:db8::1", "4401:2bc:603f:d91d:27f:ff8e:e6f1:dc1e"},
+    {"2001:db8::2", "4401:2bc:603f:d91d:27f:ff8e:e6f1:dc1c"},
+  };
+
+  for (const auto& [original, pseudonymized] : addresses) {
+    auto original_address = *to<address>(original);
+    auto pseudonymized_address_expectation = *to<address>(pseudonymized);
+
+    auto pseudonymized_adress_actual
+      = address::pseudonymized(original_address, seed);
+
+    REQUIRE_EQUAL(pseudonymized_adress_actual,
+                  pseudonymized_address_expectation);
+  }
+}
+
+TEST(IPv6 pseudonymization - seed 2) {
+  std::array<vast::address::byte_type, 32> seed
+    = {0x80, 0x09, 0xAB, 0x3A, 0x60, 0x54, 0x35, 0xBE, 0xA0, 0xC3, 0x85,
+       0xBE, 0xA1, 0x84, 0x85, 0xD8, 0xB0, 0xA1, 0x10, 0x3D, 0x65, 0x90,
+       0xBD, 0xF4, 0x8C, 0x96, 0x8B, 0xE5, 0xDE, 0x53, 0x83, 0x6E};
+  // test data from
+  // https://github.com/noinkling/node-cryptopan/blob/main/src/test/test_data.ts
+  std::unordered_map<std::string, std::string> addresses = {
+    {"2a02:0db8:85a3:0000:0000:8a2e:0370:7344", "1482:f447:75b3:f1f9:fbdf:622e:"
+                                                "34f:ff7b"},
+    {"2a02:db8:85a3:0:0:8a2e:370:7344", "1482:f447:75b3:f1f9:fbdf:622e:34f:"
+                                        "ff7b"},
+    {"2a02:db8:85a3::8a2e:370:7344", "1482:f447:75b3:f1f9:fbdf:622e:34f:ff7b"},
+    {"2a02:0db8:85a3:08d3:1319:8a2e:0370:7344", "1482:f447:75b3:f904:c1d9:ba2e:"
+                                                "489:1346"},
+    {"2001:b8:a3:00:00:2e:70:44", "1f18:b37b:1cc3:8118:41f:9fd1:f875:fab8"},
+    {"fc00::", "f33c:8ca3:ef0f:e019:e7ff:f1e3:f91f:f800"},
+  };
+
+  for (const auto& [original, pseudonymized] : addresses) {
+    auto original_address = *to<address>(original);
+    auto pseudonymized_address_expectation = *to<address>(pseudonymized);
+
+    auto pseudonymized_adress_actual
+      = address::pseudonymized(original_address, seed);
+
+    REQUIRE_EQUAL(pseudonymized_adress_actual,
+                  pseudonymized_address_expectation);
+  }
+}
+
+TEST(IPv6 pseudonymization - seed 3) {
+  std::array<vast::address::byte_type, 32> seed
+    = {0,  1,  2,  3,  4,  5,  6,  7,  8,  9,  10, 11, 12, 13, 14, 15,
+       16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31};
+  // test data from
+  // https://github.com/noinkling/node-cryptopan/blob/main/src/test/test_data.ts
+  std::unordered_map<std::string, std::string> addresses = {
+    {"2001:db8::1", "dd92:2c44:3fc0:ff1e:7ff9:c7f0:8180:7e00"},
+  };
+
+  for (const auto& [original, pseudonymized] : addresses) {
+    auto original_address = *to<address>(original);
+    auto pseudonymized_address_expectation = *to<address>(pseudonymized);
+
+    auto pseudonymized_adress_actual
+      = address::pseudonymized(original_address, seed);
+
+    REQUIRE_EQUAL(pseudonymized_adress_actual,
+                  pseudonymized_address_expectation);
+  }
```

Comment:
This exact same code occurs a few times in this file. Should we move it into a free function in an anonymous namespace?

Also, as per the style guide, please remove the whitespace between the lines within functions.